src/vm/diagnosticserver.cpp

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "common.h"
+#include "diagnosticserver.h"
+#include "diagnosticsipc.h"
+#include "eventpipeprotocolhelper.h"
+
+#ifdef FEATURE_PAL
+#include "pal.h"
+#endif // FEATURE_PAL
+
+#ifdef FEATURE_PERFTRACING
+
+static DWORD WINAPI DiagnosticsServerThread(LPVOID lpThreadParameter)
+{
+    CONTRACTL
+    {
+        // TODO: Maybe this should not throw.
+        THROWS;
+        GC_TRIGGERS;
+        MODE_ANY;
+        PRECONDITION(lpThreadParameter != nullptr);
+    }
+    CONTRACTL_END;
+
+    auto pIpc = reinterpret_cast<IpcStream::DiagnosticsIpc *>(lpThreadParameter);
+    if (pIpc == nullptr)
+    {
+        STRESS_LOG0(LF_STARTUP, LL_ERROR,"Diagnostics IPC listener was undefined\n");
+        return 1;
+    }
+
+#ifdef _DEBUG
+    ErrorCallback LoggingCallback = [](const char *szMessage, uint32_t code) {
+        LOG((LF_REMOTING, LL_WARNING, "warning (%d): %s.\n", code, szMessage));
```

Comment:
You can open an issue and fix this later:
1) LF_REMOTING is referring to the .NET remoting feature. We probably want a different (new) flag to use here.
2) I'd recommend using STRESS_LOG rather than LOG so that the failures are diagnosable in retail builds.

Reply:
I'm only logging client connection problems in DEBUG. Is this something we want on retail too (then I can remove the `#if _DEBUG`)?

---
In reply to: [265420929](https://github.com/dotnet/coreclr/pull/23106#discussion_r265420929) [](ancestors = 265420929)

Reply:
yeah, I would also remove the #ifdef DEBUG (as part of the issue you fix later)